packages/ember-routing-htmlbars/lib/keywords/action.js

```diff
@@ -8,60 +8,151 @@ import { keyword } from 'htmlbars-runtime/hooks';
+    };
+  })(context);
+  div.onclick = actionFunction;
+  ```
+
+  Thus when the div is clicked, the action on that context is called.
+  Becuase the `actionFunction` is just a function, closure actions can be
+  passed between components the still execute in the correct context.
+
+  Here is an example action handler on a component:
+
+  ```js
+  export default Ember.Component.extend({
     actions: {
-      anActionName: function() {
+      save(/* event *\/) {
+        this.get('model').save();
       }
     }
   });
   ```
 
-  Will result in the following rendered HTML
+  Actions are always looked up on the `actions` property of the current context.
+  This avoids collisions in the naming of common actions, such as `destroy`.
 
-  ```html
-  <div class="ember-view">
-    <div data-ember-action="1">
-      click me
-    </div>
-  </div>
+  Two options can be passed to the `action` helper when it is used in this way.
+
+  * `target=someProperty` will look to `someProperty` instead of the current
+    context for the `actions` hash. This can be useful when targetting a
+    service for actions.
+  * `value="target.value"` will read the path `target.value` off the first
+    argument to the action when it is called and rewrite the first argument
+    to be that value. This is useful when attaching actions to event listeners.
+
+  ### Invoking an action
+
+  Closure actions curry both their scope and any arguments. When invoked, any
+  additional arguments are added to the already curried list.
+
+  Actions should be invoked using the [sendAction](/api/classes/Ember.Component.html#method_sendAction)
+  method. The first argument to `sendAction` is the action to be called, and
+  additional arguments are passed to the action function. This has interesting
+  properties combined with currying of arguments. For example:
+
+  ```js
+  export default Ember.Component.extend({
+    actions: {
+      // Usage {{input on-input=(action (action 'setName' model) value="target.value")}}
+      setName(model, name) {
+        model.set('name', name);
+      }
+    }
+  });
+  ```
+
+  The first argument (`model`) was curried over, and the run-time argument (`event`)
+  becomes a second argument. Action calls be nested this way because each simply
+  returns a function. Any function can be passed to the `{{action` helper, including
+  other actions.
+
+  Actions invoked with `sendAction` have the same currying behavior as demonstrated
+  with `on-input` above. For example:
+
+  ```js
+  export default Ember.Component.extend({
+    actions: {
+      setName(model, name) {
+        model.set('name', name);
+      }
+    }
+  });
   ```
 
-  Clicking "click me" will trigger the `anActionName` action of the
-  `App.ApplicationController`. In this case, no additional parameters will be passed.
+  ```handlebars
+  {{my-input submit=(action 'setName' model)}}
+  ```
 
-  If you provide additional parameters to the helper:
+  ```js
+  // app/component/my-component.js
```

Comment:
Should be `app/components` (plural), and like use a name that isn't ambiguous with the type.